daemon/mount.c

```diff
@@ -142,6 +142,38 @@ mount_get_entry_by_img(const mount_t *mnt, const char *img)
 	return NULL;
 }
 
+uint64_t
+mount_get_disk_usage(const mount_t *mnt)
+{
+	ASSERT(mnt);
+	size_t count = mount_get_count(mnt);
+	uint64_t disk_usage = 0;
+
+	for (size_t i = 0; i < count; i++) {
+		// todo: filter types
+		mount_entry_t *entry = mount_get_entry(mnt, i);
+		switch (entry->type) {
+		case MOUNT_TYPE_DEVICE:
+		case MOUNT_TYPE_DEVICE_RW:
+		case MOUNT_TYPE_EMPTY:
+		case MOUNT_TYPE_COPY:
+		case MOUNT_TYPE_OVERLAY_RW: {
+			if (disk_usage <= (UINT64_MAX - entry->image_size)) {
+				disk_usage += entry->image_size;
```

Comment:
in c_vol.c internally also a meta.img with `size = entry->image_size / 10` is created.
You have to add this here, too.
It would make sense to introduce a define in mount.h for MOUNT_DM_INTEGRITY_META_FACTOR or something like that. And also use this in c_vol.c.

Reply:
Ok. This meta.img is only created for mount types SHARED, SHARED_RW, OVERLAY_RW and EMPTY right?

Reply:
actually only for SHARED_RW, OVERLAY_RW and EMPTY, but it would make sence to extend c_vol
to also use integrity for TYPE_COPY and TYPE_DEVICE_RW, but this we can postpone to another PR.

Reply:
Added support to handle this case for OVERLAY_RW and EMPTY.